### PR TITLE
Feature: Added ability to toggle auto assignment of conversations

### PR DESCRIPTION
**Pull request by venarius**
# Pull Request Template

## Description

This adds the ability for admins to enable or disable the round robin of conversations. I wanted to add a feature instead of just writing tests. I hope that's okay 😅 But will continue to write tests of course 😀

Do I also need to add this to a documentation?

For #489 

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Added new tests for it.

### CODE REVIEW

File: app/javascript/dashboard/store/modules/inboxes.js [hunk 1]
```diff
@@ -76,6 +76,13 @@ export const actions = {
       throw new Error(error);
     }
   },
+  updateAutoAssignment: async (nil, { id, ...inboxParams }) => {
+    try {
+      await InboxesAPI.update(id, inboxParams);
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/inboxes.js]: Please update the local state as well. You should be sending `EDIT_INBOXES` mutation

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue [hunk 2]
```diff
@@ -82,8 +96,10 @@ export default {
   data() {
     return {
       selectedAgents: [],
+      autoAssignment: false,
       isUpdating: false,
       isAgentListUpdating: false,
+      isAutoAssignmentUpdating: false,
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue]: I would recommend to move this out of the the component, please see the uiFlags usage in the inboxes module

**Comment by pranavrajs** [comment]
@venarius Nice work ❤️Please the inline comments.

File: app/javascript/dashboard/store/modules/inboxes.js [hunk 3]
```diff
@@ -76,10 +77,14 @@ export const actions = {
...
     }
   },
-  updateAutoAssignment: async (nil, { id, ...inboxParams }) => {
+  updateAutoAssignment: async ({ commit }, { id, ...inboxParams }) => {
+    commit(types.default.SET_INBOXES_UI_FLAG, { isUpdatingAutoAssignment: true });
     try {
-      await InboxesAPI.update(id, inboxParams);
+      const response = await InboxesAPI.update(id, inboxParams);
+      commit(types.default.EDIT_INBOXES, response.data);
+      commit(types.default.SET_INBOXES_UI_FLAG, { isUpdatingAutoAssignment: false });
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/inboxes.js]: Please update the test cases for the same :)

Comment (venarius) [reply app/javascript/dashboard/store/modules/inboxes.js]: Yes, I am currently doing this :)

**Comment by venarius** [comment]
@pranavrajs all done :)

**Comment by ReactRaja** [comment]
Still cant find the option for auto-assignment in super admin panel. Is there any undocumented env_variable which i need to enable for this to appear?

**Comment by pranavrajs** [comment]
@ReactRaja Please take a look at Settings -> Inboxes -> settings

Screenshot 2020-12-14 at 7 11 48 PM

**Comment by ReactRaja** [comment]
Huh ok, its enabled for me but wasn't assigning automatically. Guess ill reinstall it . 
Keep up the good work guys and thanks for the support

**Pull request merged**